### DISCUSSION TITLE
how to pass space-separated args to an external command?

### DISCUSSION THREAD

**Discussion by JeffMelton** [nushell/nushell#11806] [Q&A] [answered]
I use an external command that can accept space-separated args, e.g., `<command> <action> foo bar baz`. It executes `<action>` across each of the args.

I'm collecting those args from a table and joining them on a space character: `let args = $table.id | str join ' '`. I shell that out to bash: `bash -c $"<command> <action> ($args)"` because if I do `<command> <action> $args` in nushell, `<command>` sees `$args` as a single argument. I've twiddled around with lots of ideas, but haven't found a way yet to do it in nushell natively. Can it be done?

**Comment by fdncred** [comment, marked as answer]
I think what you want to do is collect those arguments in a list like
```nushell
let args = [-a --arg2 blah something else]
```
and then call your external with `$args` or maybe `...$args`.

**Reply by JeffMelton** [reply]
Ah, okay, I'm making it harder than it needs to be. Thanks!
